tests/test_machine.py

```diff
@@ -27,23 +27,23 @@ def test_wrong_command(self) -> None:
 
         # Check = False to see if stderr and returncode have the expected values
         result = self.machine.run(["NonExisting", "Command"], check=False)
-        assert result.returncode in (203, 127)
+        assert result.returncode in (1, 127, 203)
 
         result = self.machine.run(["ls", "-"], check=False)
-        assert result.returncode == 2
+        assert result.returncode in (1, 2)
         assert "No such file or directory" in result.stderr
 
     def test_infinite_command(self) -> None:
         with pytest.raises(TimeoutExpired):
             self.machine.run(["tail", "-f", "/dev/null"], 2)
 
-
+@pytest.mark.skipif(os.getenv("MKOSI_TEST_DEFAULT_VERB") == "shell", reason="Shell never boots the machine.")
```

Comment:
I think it's more robust to check the actual verb of the machine instead of checking the environment variable. You can check the verb inside the test and call pytest.skip() to skip the test if it is shell.